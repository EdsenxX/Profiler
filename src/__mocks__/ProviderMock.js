import React from 'react';
import { createStore } from 'redux';
import { Router} from 'react-router-dom';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import reducers from '../reducers';

const store = createStore(reducers, {});
const history = createBrowserHistory();

const ProviderMock = props => (
  <Provider store={store}>
    <Router location={history.location} navigator={history}>
      {props.children}
    </Router>
  </Provider>
)

export default ProviderMock;