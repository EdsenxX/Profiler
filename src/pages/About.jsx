// Components
import Layout from "../components/Layout";
// Assets
import imageProfile from "../assets/images/yo.jpg";

const About = () => {
  return (
    <Layout>
      <div className="w-full flex justify-center mt-8 mb-16">
        <div className="w-4/5 flex justify-center">
          <div className="w-[350px] bg-white rounded-xl p-7">
            <div className="flex flex-col items-center">
              <img
                src={imageProfile}
                alt="Eduardo Serrano Jaime"
                className="rounded-full w-[170px] mb-2"
              />
              <h1>Eduardo Serrano Jaime</h1>
              <p className="text-sm">JavaScript Developer</p>
            </div>
            <div className="flex gap-2 justify-center text-xl my-2">
              <a href="https://github.com/EdsenxX" rel="noreferrer">
                <i className="fa-brands fa-github"></i>
              </a>
              <a href="https://www.linkedin.com/in/edsen/" rel="noreferrer">
                <i className="fa-brands fa-linkedin"></i>
              </a>
              <a href="https://edsen.dev" target="_blank" rel="noreferrer">
                <i className="fa-solid fa-earth-americas"></i>
              </a>
            </div>
            <p className="text-md text-center">
              Soy Javascript Developer con +3 años de experiencia trabajando
              profesionalmente. Tengo experiencia trabajando con instituciones
              publicas, consultoras de software, startups y también como freelancer
            </p>
          </div>
        </div>
      </div>
    </Layout>
  );
};

export default About;
