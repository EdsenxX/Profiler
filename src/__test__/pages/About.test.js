import React from "react";
import { shallow, mount } from "enzyme";
import About from "../../pages/About";
import ProviderMock from "../../__mocks__/ProviderMock";
import {create} from "react-test-renderer";

describe("About", () => {
  test("Render del componente About", () => {
    const about = shallow(
      <ProviderMock>
        <About />
      </ProviderMock>
    );
    expect(about.length).toEqual(1);
  });
});

describe("About snapshot", () => {
  test("Comprobar la UI del componente About", () => {
    const about = create(
      <ProviderMock>
        <About/>
      </ProviderMock>
    )
    expect(about.toJSON()).toMatchSnapshot();
  });
});