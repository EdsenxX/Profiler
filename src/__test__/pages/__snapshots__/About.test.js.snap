// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About snapshot Comprobar la UI del componente About 1`] = `
<div>
  <nav
    className="text-white flex p-5 bg-primary-100 items-center shadow-md shadow-secondary-100/30 justify-between"
  >
    <h1
      className="font-title text-4xl"
      id="Navbar-title"
    >
      Profiler
    </h1>
    <div
      className="flex gap-4 text-lg"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </div>
  </nav>
  <div
    className="w-full flex justify-center mt-8 mb-16"
  >
    <div
      className="w-4/5 flex justify-center"
    >
      <div
        className="w-[350px] bg-white rounded-xl p-7"
      >
        <div
          className="flex flex-col items-center"
        >
          <img
            alt="Eduardo Serrano Jaime"
            className="rounded-full w-[170px] mb-2"
            src="yo.jpg"
          />
          <h1>
            Eduardo Serrano Jaime
          </h1>
          <p
            className="text-sm"
          >
            JavaScript Developer
          </p>
        </div>
        <div
          className="flex gap-2 justify-center text-xl my-2"
        >
          <a
            href="https://github.com/EdsenxX"
            rel="noreferrer"
          >
            <i
              className="fa-brands fa-github"
            />
          </a>
          <a
            href="https://www.linkedin.com/in/edsen/"
            rel="noreferrer"
          >
            <i
              className="fa-brands fa-linkedin"
            />
          </a>
          <a
            href="https://edsen.dev"
            rel="noreferrer"
            target="_blank"
          >
            <i
              className="fa-solid fa-earth-americas"
            />
          </a>
        </div>
        <p
          className="text-md text-center"
        >
          Soy Javascript Developer con +3 años de experiencia trabajando profesionalmente. Tengo experiencia trabajando con instituciones publicas, consultoras de software, startups y también como freelancer
        </p>
      </div>
    </div>
  </div>
</div>
`;
